packages/cdk8s/README.md

```diff
@@ -76,6 +76,84 @@ NOTE: names of included objects (`metadata.name`) are preserved. This means that
 if you try to include the same manifest twice into the same chart, your manifest
 will have duplicate definitions of the same objects.
 
+### Dependencies
+
+You can declare dependencies between various `cdk8s` constructs by using the built-in support of the underlying `constructs` model.
+
+#### `ApiObject`s
+
+For example, you can force kubernetes to first apply a `Service` before applying the `Deployment` that uses that service.
+
+```typescript
+
+const deployment = new k8s.Deployment(chart, 'Deployment');
+const service = new k8s.Service(chart, 'Service');
+
+Node.of(deployment).addDependency(service);
+```
+
+`cdk8s` will ensure that the `service` construct is placed before the `deployment` construct in the resulting manifest:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: cdk8s-playground-service-f42ff561
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cdk8s-playground-deployment-bef88ab7
+```
+
+#### `Chart`s
+
+You can also specify dependencies between charts, in exactly the same manner. For example, if we have a chart that deploys our `namespace`, we need that chart to be applied first:
+
+```typescript
+const namespaceChart = new MyChart(app, 'namespace');
+const applicationChart = new MyChart(app, 'application');
+
+Node.of(applicationChart).addDependency(namespaceChart);
+```
+
+Running `cdk8s synth` will produce the following dist directory:
+
+```console
+> cdk8s synth
+
+dist/0000-namespace.k8s.yaml
+dist/0001-application.k8s.yaml
+```
+
+Notice that the `namespace` chart appears first with the `0000` prefix. This will ensure that a subsequent execution of `kubectl apply -f dist/` will apply the `namespace` first, and the `application` second.
+
+#### Things just got cool
+
+If you simply declare a dependency between two `ApiObject`s, that belong to two different `Chart`s, `cdk8s` will create the chart dependency automatically for you.
+
+```typescript
+const namespaceChart = new MyChart(app, 'namespace');
+const applicationChart = new MyChart(app, 'application');
+
+const namespace = new k8s.Namespace(namespaceChart, 'namespace');
+const deployment = new k8s.Deployment(applicationChart, 'Deployment');
+
+// dependency between ApiObjects, not Charts!
+Node.of(deployment).addDependency(namespace);
+```
+
+Running `cdk8s synth` will produce the same result as if explicit chart dependencies were declared:
+
+```console
+> cdk8s synth
+
+dist/0000-namespace.k8s.yaml
+dist/0001-application.k8s.yaml
+```
+
+This means you need not be bothered with managing chart dependencies, simply work with the `ApiObject`s you create, and let `cdk8s` infer the chart dependencies.
+
```

Comment:
We still don’t support deps between custom constructs correct? Worth mentioning and referencing an issue

Reply:
Adding it